### DISCUSSION TITLE
Incorrect location of Group added after centering and scaling

### DISCUSSION THREAD

**Discussion by 767339608** [fabricjs/fabric.js#7191] [Q&A]
Add Group after zooming in and out. I take the current position of the element minus the shifted offset. The result is removed to scale. I found the wrong value. Is there any way to solve this problem

**Comment by 767339608** [comment]
For example
 canvas.on({
            "mouse:down": (e: any) => {
                if (cursorStatus) {
                    panning = true;
                    moveImgX = 0
                    moveImgY = 0
                    canvas.selection = false;
                }
            },
            "mouse:up": (e: any) => {
                if (cursorStatus) {
                    //Migration accumulation
                    setimgY(imgY => imgY += moveImgY)
                    setimgX(imgx => imgx += moveImgX)
                    panning = false;
                    canvas.selection = true;
                }
            },
            "mouse:move": (e: any) => {
                if (panning && e && e.e && cursorStatus) {
                    let delta = new fabric.Point(e.e.movementX, e.e.movementY);
                    moveImgX += delta.x
                    moveImgY += delta.y
                    canvas.relativePan(delta);
                }
            },
            "object:selected": (e: any) => {
                console.log(e)
            }
        })
const center = canvas.getCenter()
//Imgscale is the scale
let x=(center.left - imgX)/imgScale
let Y=center.top- imgY/imgScale

**Reply by i-prikot** [reply]
let Y=center.left - imgY/imgScale

Why `center.left` and not `center.top`? 
Why `let Y=center.left - imgY/imgScale` and not `let Y=(center.left - imgY)/imgScale`?

You can create an example in https://jsfiddle.net/?

**Reply by 767339608** [reply]
Which one is the wrong copy I'm sorry. The example is already in the codeSandbox
https://codesandbox.io/s/quirky-snow-xdl9j?file=/src/App.tsx

**Reply by i-prikot** [reply]
Set origin and position for group. 
`originY: "center"`,   `originX: "center"`,` left: xaxis`,   `top: yaxis`

```
let Group = new fabric.Group([circle, text], {
      left: xaxis,
      top: yaxis,
      objectCaching: false,
      hasControls: false,
      lockRotation: true,
      originY: "center",
      originX: "center"
});
```

**Reply by 767339608** [reply]
When zooming in, the dot hits the outside of the display

**Reply by 767339608** [reply]
Because the viewing area is enlarged

**Reply by i-prikot** [reply]
> When zooming in, the dot hits the outside of the display

Make the canvas as wide as the window area.
